### PR TITLE
WIP: Summary stats

### PR DESCRIPTION
**Pull request by WestfalNamur**
summary_stats branch to upstream-master

### CODE REVIEW

File: webviz_subsurface/containers/_summary_stats.py [hunk 1]
```diff
@@ -59,62 +91,90 @@ def layout(self):
...
+    Dash functions take positional args., so order matters. """
 
-    data = get_summary_data(ensemble_path,
-                            sampling)[['REAL', 'DATE', vector]]
+    df_ens_set = []
+
+    for path in ensemble_paths:
+        stats = scratch_ensemble('', path).get_smry_stats(
+                time_index=sampling, column_keys=column_keys)
+        stats['ENS'] = path.replace(
```
Comment (HansKallekleiv) [inline webviz_subsurface/containers/_summary_stats.py]: 'ENS' should be 'ENSEMBLE' as this is the standard column name

File: webviz_subsurface/containers/_summary_stats.py [hunk 2]
```diff
@@ -59,62 +91,90 @@ def layout(self):
...
+    """ returns a single dcc.Graph """
+
+    summary_stats = get_summary_data(ensemble_paths, column_keys, sampling
+                                     )[['REAL', 'DATE', 'ENS', vector]]
 
     traces = [{
         'x': df['DATE'],
         'customdata': df['REAL'],
         'y': df[vector],
         'name': name,
```
Comment (HansKallekleiv) [inline webviz_subsurface/containers/_summary_stats.py]: *name* should be the ensemble name not the ensemble path

File: webviz_subsurface/containers/_summary_stats.py [hunk 3]
```diff
@@ -59,62 +91,90 @@ def layout(self):
...
         'type': 'line'
-    } for name, df in data.groupby('REAL') if name != 'DATE']
+    } for name, df in summary_stats.groupby('ENS')]
```
Comment (HansKallekleiv) [inline webviz_subsurface/containers/_summary_stats.py]: *ENS* should be *ENSEMBLE*

File: webviz_subsurface/containers/_summary_stats.py [hunk 4]
```diff
@@ -59,62 +91,90 @@ def layout(self):
...
+
+
+@cache.memoize(timeout=cache.TIMEOUT)
+def render_realization_plot(ensemble_paths, sampling, column_keys, vector):
+    """ returns a single dcc.Graph """
+
+    summary_stats = get_summary_data(ensemble_paths, column_keys, sampling
+                                     )[['REAL', 'DATE', 'ENS', vector]]
 
     traces = [{
```
Comment (HansKallekleiv) [inline webviz_subsurface/containers/_summary_stats.py]: There's a bug here that plots missing realizations as a trace.
Test using e.g. : /scratch/fmu/sago/3_r001_reek/realization-*/iter-4

img

File: webviz_subsurface/containers/_summary_stats.py [hunk 5]
```diff
@@ -59,62 +91,90 @@ def layout(self):
...
-                                     'statistics': False}]),
-                (get_summary_data, [{'ensemble_path': self.ensemble_path,
-                                     'sampling': self.sampling,
-                                     'statistics': True}])]
+        return [(get_summary_data, [{'ensemble_paths': self.ensemble_paths,
+                                     'column_keys': self.column_keys,
+                                     'sampling': self.sampling}]),
+                (get_summary_stats, [{'ensemble_paths': self.ensemble_paths,
+                                      'column_keys': self.column_keys,
+                                      'sampling': self.sampling}])]
```
[the remark this answers is not in the document]
Comment (WestfalNamur) [reply webviz_subsurface/containers/_summary_stats.py]: Does the store not take any lists at all? 
ensemble_paths and column_keys are lists them self?

**Comment by WestfalNamur** [comment]
Travis fails with:
E       selenium.common.exceptions.SessionNotCreatedException: Message: session not created: Chrome version must be between 70 and 73
E         (Driver info: chromedriver=73.0.3683.68 (47787ec04b6e38e22703e856e101e840b65afe72),platform=Linux 4.4.0-101-generic x86_64)

My chromium version is: 73.0.3683.75
Where does travis run the code?

**Comment by WestfalNamur** [comment]
Frozen for now.
Wait for PR "Update chrome driver used in Travis" 
Current state:
Summary-plot works locally (tested on Reek data) but Travis tests fails.

**Comment by anders-kiaer** [comment]
Looks like this has been moved to https://github.com/equinor/webviz-subsurface/pull/24, let me know if this PR should be opened again.

**Pull request closed without merging**